server/src/main/java/org/elasticsearch/action/support/nodes/TransportNodesAction.java

```diff
@@ -82,7 +82,7 @@ protected TransportNodesAction(
         this.transportService = Objects.requireNonNull(transportService);
         this.finalExecutor = threadPool.executor(executor);
         this.transportNodeAction = actionName + "[n]";
-        transportService.registerRequestHandler(transportNodeAction, executor, nodeRequest, new NodeTransportHandler());
+        transportService.registerRequestHandler(transportNodeAction, this.finalExecutor, nodeRequest, new NodeTransportHandler());
```

Comment:
nit: may as well be consistent with `transportNodeAction` here:

```suggestion
        transportService.registerRequestHandler(transportNodeAction, finalExecutor, nodeRequest, new NodeTransportHandler());
```